### PR TITLE
fix(kit): `CalendarRange` not update `selectedActivePeriod`, when `value` updates

### LINKED ISSUE #8527: 🐞 - TuiCalendarRange change does not update checkbox position
### Playground Link

https://stackblitz.com/edit/angular-wm2zfa?file=src%2Fapp%2Fapp.component.ts

### Description

Hi!

When assigning a new value to the calendar's date field, the checkbox indicating the selected day does not update its position accordingly. The checkbox remains associated with the previous date instead of moving to the newly selected date.

When assigning a new value to the calendar's date field via a new object reference, the checkbox indicating the selected day does not update its position accordingly. The checkbox remains associated with the previous date instead of moving to the newly selected date. However, if the value is updated by selecting a key from the existing object (old reference), the issue does not occur, and the checkbox updates correctly.

### Angular version

17

### Taiga UI version

3.90

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by mdlufy**
Closes #8527 

After update value:

**As Is:**

https://github.com/user-attachments/assets/c6f5205d-3394-4894-9c24-ba53cb2d435f

**To Be:**

https://github.com/user-attachments/assets/8aeaab56-f254-4618-82e7-b97092da5ace

### CODE REVIEW

File: projects/kit/components/calendar-range/calendar-range.component.ts [hunk 1]
```diff
@@ -107,6 +111,10 @@ export class TuiCalendarRangeComponent implements TuiWithOptionalMinMax<TuiDay>
         }
 
         valueChanges.pipe(tuiWatch(cdr), takeUntil(destroy$)).subscribe(value => {
+            if (value?.toString() !== this.selectedActivePeriod?.range?.toString()) {
+                this.selectedActivePeriod = null;
+            }
```
Comment (splincode) [inline projects/kit/components/calendar-range/calendar-range.component.ts:116]: reuse in method

Comment (mdlufy) [reply projects/kit/components/calendar-range/calendar-range.component.ts]: done

File: projects/kit/components/calendar-range/calendar-range.component.ts [hunk 2]
```diff
@@ -121,6 +129,15 @@ export class TuiCalendarRangeComponent implements TuiWithOptionalMinMax<TuiDay>
         this.value = this.previousValue;
     }
 
+    ngOnChanges({value}: SimpleChanges): void {
+        if (
+            (value?.currentValue as TuiDayRange)?.toString() !==
+            this.selectedActivePeriod?.range.toString()
+        ) {
+            this.selectedActivePeriod = null;
+        }
```
[the remark this answers is not in the document]
Comment (mdlufy) [reply projects/kit/components/calendar-range/calendar-range.component.ts]: done

**Review by splincode — APPROVED** [review]

**Comment by waterplea** [comment]
Closing in favor of https://github.com/taiga-family/taiga-ui/pull/8595

**Pull request closed without merging**